packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+  /**
+   * If `true`, the checkbox will be initially checked.
+   *
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /**
+   * The callback invoked when the checked state of the `Checkbox` changes.
+   */
+  onChange?: (isChecked: boolean) => void;
+  /**
+   * Sets the label text of the checkbox
+   */
+  children: string;
+  /**
+   * Help text of the checkbox
+   */
+  helpText?: string;
+  /**
+   * Error text of the checkbox
+   *
+   * Renders when `validationState` is set to 'error'
+   *
+   * Overrides helpText
+   */
+  errorText?: string;
+  /**
+   * If `true`, the checkbox will be indeterminate.
+   * This does not modify the isChecked property.
+   *
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * The name of the input field in a checkbox
+   * (Useful for form submission).
+   */
+  name?: string;
+  /**
+   * The value to be used in the checkbox input.
+   * This is the value that will be returned on form submission.
+   */
+  value?: string;
+  /**
+   * If `true`, the checkbox will be disabled
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * If `true`, the checkbox input is marked as required,
+   * and `required` attribute will be added
+   *
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * If `error`, the checkbox input is marked as invalid,
+   * and `invalid` attribute will be added
+   *
+   * @default undefined
+   */
+  validationState?: 'error' | undefined;
+};
+
+const Checkbox = ({
+  defaultChecked,
+  validationState,
+  isChecked,
+  isDisabled,
+  isIndeterminate,
+  isRequired,
+  name,
+  onChange,
+  value,
+  children,
+  helpText,
+  errorText,
+}: CheckboxProps): React.ReactElement => {
+  const groupProps = useCheckboxGroupContext();
+
+  // ban controlling checkbox manually while inside group
+  const hasDefaultChecked = !isUndefined(defaultChecked);
+  const hasIsChecked = !isUndefined(isChecked);
+  const hasOnChange = !isUndefined(onChange);
+  if ((hasDefaultChecked || hasIsChecked || hasOnChange) && !isEmpty(groupProps)) {
+    const props = [
+      hasDefaultChecked ? 'defaultChecked' : undefined,
+      hasIsChecked ? 'isChecked' : undefined,
+      hasOnChange ? 'onChange' : undefined,
+    ].filter(Boolean);
+
+    throw new Error(
+      `[Blade Checkbox]: Cannot set \`${props.join(
+        ',',
+      )}\` on <Checkbox /> when it's inside <CheckboxGroup />, Please set it on the <CheckboxGroup /> itself`,
+    );
+  }
+  // mandate value prop when using inside group
+  if (!value && !isEmpty(groupProps)) {
+    throw new Error(
+      `[Blade Checkbox]: <CheckboxGroup /> requires that you pass unique "value" prop to each <Checkbox />
+      <CheckboxGroup>
+        <Checkbox value="apple">Apple</Checkbox>
+        <Checkbox value="mango">Mango</Checkbox>
+      </CheckboxGroup>
+      `,
+    );
+  }
+  // If validationState is used while inside group
+  if (validationState && !isEmpty(groupProps)) {
```

Comment:
why can't we club this with above checks?
```
if ((hasDefaultChecked || hasIsChecked || hasOnChange) && !isEmpty(groupProps)) {
```

Reply:
We can!